docs/DeveloperGuide.md

```diff
@@ -1,15 +1,126 @@
 # Developer Guide
+Welcome to the NUSPlanner Developer Guide!
+We hope this documents serves useful to understand the behind-the-scenes working of our product. 
+
+## Table of Contents
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}'
+
+## Getting started
+
+Refer to the user guide [here](https://github.com/AY2223S2-CS2113-F13-3/tp/blob/master/docs/UserGuide.md).
+
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Parser Component
+The component will return the correct command based on what the user inputs into the application.
+
+#### How is the feature implemented?
+
+The Parser component parses the command of the user input and breaks the user input into different parts based on the flags.
+This component also ensures to validate that user input is correct.
+
+The Parser component handles the following methods:
+* Add event
+* Add module
+* List event list
+* Delete event
+* Edit event
+* Find date of week
+* Format checker
+* Revise information using index
+* Revise information using name
+
+The UML diagram below illustrates the flow of how the application adds events:
+![Add Event Class Diagram](UML/Images/addEvent.png)
+
+
+The UML diagram below illustrates the flow of how the application adds modules:
+![Add Event Class Diagram](UML/Images/addModules.png)
```

Comment:
Perhaps you can include an activation bar as the object is created and for better visual understanding?
![image](https://user-images.githubusercontent.com/88631870/228732546-0e9038e8-39a5-4b09-846b-ebb83160a9f1.png)